From 38f5f31c92ea9354d40926adf884bdfe423c5622 Mon Sep 17 00:00:00 2001
From: Grzegorz Jaszczyk <jaz@semihalf.com>
Date: Mon, 22 Jul 2019 22:43:07 +0200
Subject: [PATCH 361/386] mmc: sdhci-xenon: do not switch voltage during oops

When oops_in_progress skip voltage switching since it may required
i2c blocking operation in case when vqmmc is routed via io-expanders

Change-Id: I7969dfa961a51041e43eae446e189f713e775c82
Signed-off-by: Grzegorz Jaszczyk <jaz@semihalf.com>
Reviewed-on: https://sj1git1.cavium.com/12986
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-by: Marcin Wojtas <marcin@marvell.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 drivers/mmc/host/sdhci-xenon.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/mmc/host/sdhci-xenon.c b/drivers/mmc/host/sdhci-xenon.c
index 70c82a3e8d8a..e27ad13fe026 100644
--- a/drivers/mmc/host/sdhci-xenon.c
+++ b/drivers/mmc/host/sdhci-xenon.c
@@ -325,8 +325,11 @@ static int xenon_start_signal_voltage_switch(struct mmc_host *mmc,
 	 * If Vqmmc is fixed on platform, vqmmc regulator should be unavailable.
 	 * Thus SDHCI_CTRL_VDD_180 bit might not work then.
 	 * Skip the standard voltage switch to avoid any issue.
+	 *
+	 * When oops_in_progress skip voltage switching since it may required
+	 * i2c blocking operation in case when vqmmc is routed via io-expanders.
 	 */
-	if (PTR_ERR(mmc->supply.vqmmc) == -ENODEV)
+	if (PTR_ERR(mmc->supply.vqmmc) == -ENODEV || oops_in_progress)
 		return 0;
 
 	return sdhci_start_signal_voltage_switch(mmc, ios);
-- 
2.17.1

